ev3dev2/motor.py

```diff
@@ -2099,6 +2111,169 @@ def follow_line(self,
 
         self.stop()
 
+    def calibrate_gyro(self):
+        """
+        Calibrates the gyro sensor.
+
+        NOTE: This takes 1sec to run
+        """
+        assert self.gyro, "GyroSensor must be defined"
+
+        for x in range(2):
+            self.gyro.mode = 'GYRO-RATE'
+            self.gyro.mode = 'GYRO-ANG'
+            time.sleep(0.5)
+
+    def follow_gyro_angle(self,
+            kp, ki, kd,
+            speed,
+            target_angle=0,
+            sleep_time=0.01,
+            follow_for=follow_for_forever,
+            **kwargs
+        ):
+        """
+        PID gyro angle follower
+
+        ``kp``, ``ki``, and ``kd`` are the PID constants.
+
+        ``speed`` is the desired speed of the midpoint of the robot
+
+        ``target_angle`` is the angle we want drive
+
+        ``sleep_time`` is how many seconds we sleep on each pass through
+            the loop.  This is to give the robot a chance to react
+            to the new motor settings. This should be something small such
+            as 0.01 (10ms).
```

Comment:
How does it perform with a 10ms loop time? Does it actually run at that frequency or is it limited by the execution time?

Reply:
Excellent question. My tests have primarily been to use the default of 10ms. When using the gyro in EV3-G, we've had to introduce some amount of small sleep for the gyro to chillax. This allows users to tweak this if their gyro needs more rest between samples.